StyleCop.Analyzers/StyleCop.Analyzers/Helpers/RobustFixAllProvider.cs

```diff
@@ -19,11 +35,11 @@ internal sealed class SA1407SA1408FixAllProvider : FixAllProvider
             {
             case FixAllScope.Document:
                 var newRoot = await FixAllInDocumentAsync(fixAllContext, fixAllContext.Document);
-                return CodeAction.Create("Add parentheses", fixAllContext.Document.WithSyntaxRoot(newRoot));
+                return CodeAction.Create(codeActionText, fixAllContext.Document.WithSyntaxRoot(newRoot));
 
             case FixAllScope.Project:
                 Solution solution = await GetProjectFixesAsync(fixAllContext, fixAllContext.Project);
-                return CodeAction.Create("Add parentheses", solution);
+                return CodeAction.Create(codeActionText, solution);
 
             case FixAllScope.Solution:
```

Comment:
I'm not confident our handling of solutions supports cases where a single file is included in multiple projects, and the result of applying the fix to a document in one project differs from the result of applying the fix in another project.

For any two projects P1 and P2 which contain the same document D, the result of applying the code fix to D in P1 should be identical to the result of applying the code fix to D in P2, with the following exception. Without loss of generality, assume P1 contains a span of text T which is part of the syntax, but the same span T in P2 is `DisabledTextTrivia`. For the span T:

1. The code fix to D in P2 is expected to not change the span (i.e. no code fix should make any changes to a `DisabledTextTrivia` trivia).
2. For the span T, the code fix to D in P1 is used to produce the final change to the document D.